src/openrct2/network/Network.cpp

```diff
@@ -1625,9 +1625,10 @@ uint8_t* Network::save_for_network(size_t& out_size, const std::vector<const Obj
     const void* data = ms.GetData();
     int32_t size = ms.GetLength();
 
-    uint8_t* compressed = util_zlib_deflate(static_cast<const uint8_t*>(data), size, &out_size);
-    if (compressed != nullptr)
+    auto compressed = util_zlib_deflate(static_cast<const uint8_t*>(data), size);
+    if (compressed != std::nullopt)
```

Comment:
Can't we just return the vector and just use `.empty()`?

```suggestion
    auto compressed = util_zlib_deflate(static_cast<const uint8_t*>(data), size);
    if (!compressed.empty())
```

Reply:
I don't think that is a good idea. What happens if we compress zero data?

Reply:
Can't we just not call it if `size == 0`?

Reply:
Std optional is meant to be zero overhead so there should be no penalty using it. It makes it clear that this function has a failure state that must be checked.

Reply:
Alright, I just wanted to be sure we were not overusing it.